File: api/src/main/scala/com/avast/clients/rabbitmq/api/Delivery.scala
package com.avast.clients.rabbitmq.api

import com.avast.bytes.Bytes

sealed trait Delivery[+A] {
  def properties: MessageProperties

  def routingKey: String
}

object Delivery {

  case class Ok[+A](body: A, properties: MessageProperties, routingKey: String) extends Delivery[A]

  case class MalformedContent(body: Bytes, properties: MessageProperties, routingKey: String, ce: ConversionException)
      extends Delivery[Nothing]

  def apply[A](body: A, properties: MessageProperties, routingKey: String): Delivery.Ok[A] = {
    Delivery.Ok(body, properties, routingKey)
  }
}

File: api/src/main/java/com/avast/clients/rabbitmq/javaapi/Delivery.java
package com.avast.clients.rabbitmq.javaapi;

import com.avast.bytes.Bytes;

public class Delivery {
    private final Bytes body;
    private final String routingKey;
    private final MessageProperties properties;


    public Delivery(final String routingKey, final Bytes body, final MessageProperties properties) {
        this.routingKey = routingKey;
        this.body = body;
        this.properties = properties;
    }

    public Bytes getBody() {
        return body;
    }

    public String getRoutingKey() {
        return routingKey;
    }

    public MessageProperties getProperties() {
        return properties;
    }
}
